--- drivers/platform/x86/thinkpad_acpi.c.org	2016-01-06 22:03:25.012201143 +0700
+++ drivers/platform/x86/thinkpad_acpi.c	2016-01-06 22:04:38.357205225 +0700
@@ -5281,6 +5281,8 @@
 	"tpacpi::unknown_led2",
 	"tpacpi::unknown_led3",
 	"tpacpi::thinkvantage",
+	"tpacpi::unknown_led4",
+	"tpacpi::micmute",
 };
 #define TPACPI_SAFE_LEDS	0x1081U
 
@@ -5505,7 +5507,7 @@
 	{ /* Lenovo */
 	  .vendor = PCI_VENDOR_ID_LENOVO,
 	  .bios = TPACPI_MATCH_ANY, .ec = TPACPI_MATCH_ANY,
-	  .quirks = 0x1fffU,
+	  .quirks = 0x5fffU,
 	},
 	{ /* IBM ThinkPads with no EC version string */
 	  .vendor = PCI_VENDOR_ID_IBM,
